### ISSUE TITLE
Improve length and syntax restrictions for github sourced package documentation

### ISSUE THREAD

**Issue by madelson** [NuGet/NuGetGallery#5682]
I was excited to see the new feature on NuGet.org where you can import your documentation file from Github as part of uploading a package.

However, the two times I've attempted to use this feature I've been blocked by its limitations:

* I tried to import https://github.com/madelson/DistributedLock/blob/master/README.md but got an error message saying the file was too large
* I tried to import https://github.com/madelson/Bits/blob/a9739c81972902eadf06435dca7fb333c8d59b4b/README.md but the markdown table syntax didn't display correctly

I'd love to use this feature, but right now it seems too limited.

**Comment by chenriksson** [comment]
Thanks for the feedback, @madelson !

> I tried to import https://github.com/madelson/DistributedLock/blob/master/README.md but got an error message saying the file was too large

I believe we set an upper limit (currently 8KB) based on sample readmes that we found in various package repositories. We could certainly increase this... any suggestion? :-)

> I tried to import https://github.com/madelson/Bits/blob/a9739c81972902eadf06435dca7fb333c8d59b4b/README.md but the markdown table syntax didn't display correctly

We're using the CommonMark library, which does not support tables since they are not part of the core markdown spec. Unfortunately, we also disabled embedded HTML (another option for tables) due to time limitations when trying to address security concerns. Adding support for tables is something we would love to do, but will need to be prioritized with the rest of our backlog.

We do have loftier ideas for this feature, including the ability to push documentation with your package(s) or have a common documentation for all versions. Feel free to read more and/or provide feedback in the issues below:
- https://github.com/NuGet/Home/wiki/Package-README.md-support
- https://github.com/NuGet/Home/issues/6056
- https://github.com/NuGet/NuGetGallery/issues/4772

**Comment by madelson** [comment]
Thanks for the follow-up @chenriksson.

> I believe we set an upper limit (currently 8KB) based on sample readmes that we found in various package repositories. We could certainly increase this... any suggestion? :-)

I've seen many cases (besides my own) where smaller projects use the README file as a one-stop shop for documentation, which easily breaks the 8K limit but without being enormous. Larger projects tend to have enough documentation that they split it across pages. I don't know what the motivation is behind that particular limit, but it feels like the limit could be quite a bit larger 250K? without causing issues looks like the example I have is 59K.

I suppose an alternative would be to highlight the "Project Site" link (if available) much more so than it is today to make it easy for less-frequent users to locate the docs.

> We're using the CommonMark library, which does not support tables since they are not part of the core markdown spec.

That's fair, but since people are likely importing from github it feels like this feature would work much better if it supported github's flavor of markdown. Tables certainly feel like a pretty common element in documentation I've read.

**Comment by karann-msft** [comment]
Note to self - impacts https://github.com/NuGet/Home/issues/4628

**Comment by lyndaidaii** [comment]
close it that we support table now

**Comment by madelson** [comment]
@lyndaidaii great to hear. Do you know if the 8kb limit was upped?
